### ISSUE TITLE
feat: get inferred structure of a body without creating a dataset

### ISSUE THREAD

**Issue by b5** [qri-io/qri#1532]
@aborruso brought up a great idea in [chat on discord](https://discordapp.com/channels/497052800778502152/497052802275737611/756148792209703033):

> A lot of a time, working with data in csv format, it's useful to write some doc or same email in which it's necessary to add a schema.
Because CSV does not have it by default.
> For this kind of jobs, I use one of the tools of my box. For this kind of job I use goodtables or xsv. To have it in qri, it would be great, because it would become more and more a data swiss knife, to do great and proper things as create properly described dataset, but also a tool to do in right way small things with a CSV

I ❤️ whenever someone sees qri ad a "data swiss [army] knife", and think this is a great example. We should definitely have a way of getting an inferred schema without creating a version. In chat I'd proposed adding this functionality to `get` by adding the save flags to get to "hot build" a dataset:

```
$ qri get schema --body file.csv
```

**Comment by dustmop** [comment]
What's interesting is that `get` already has a need for this "hot build" functionality. If a dataset is checked out in a working directory, and has a `body.csv` file but no `structure.json`, we infer a schema in order to read the body file. This is currently handled by `fsi.GetBody` in `fsi/body.go`, and there's a long-standing TODO in `lib.Get` about needing to handle this special case, and how the code path could be simplified. If we generalize this functionality so that `get` can infer a schema in a modular fashion, both problems will get solved.

The biggest question for me is how exactly to handle the UI, and whether there's use cases that would be covered aside from getting the structure / schema. Right now `get` does not have a `--body` flag, and this would be the first time adopting it from `save`. What would this change imply? Can I override other components or just the body? Does this handle the use case of wanting to test a different schema against the same body to catch validation errors? Would like to sort that out before making a final decision.

**Comment by b5** [comment]
Agreed. I think the next step here is to use our new spec site to define the exact role of "get", and use that as a place to examine how this fits with existing tech.
